### PR TITLE
feat(notification-center): Add LogEvent notification

### PR DESCRIPTION
**Pull request by mnoman09**
Summary
-------

- Adds LogEvent notification in NotificationCenter.
- Integrated it in BatchEventDispatcher.

Test plan
---------

- Added unit tests.

### CODE REVIEW

File: tests/test_notification_center.py [hunk 1]
```diff
@@ -219,6 +235,7 @@ def test_clear_all_notification_listeners(self):
         test_notification_center.clear_all_notification_listeners()
 
         for notification_type in notification_center.NOTIFICATION_TYPES:
+            print(notification_type)
```
Comment (mjc1283) [inline tests/test_notification_center.py]: Remove

File: optimizely/event/event_processor.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 from optimizely import logger as _logging
 from optimizely.closeable import Closeable
 from optimizely.event_dispatcher import EventDispatcher as default_event_dispatcher
-from optimizely.helpers import validator
+from optimizely.helpers import validator, enums
```
Comment (aliabbasrizvi) [inline optimizely/event/event_processor.py]: nit. I'd recommend splitting up imports into separate lines. Just a conventional thing.

File: optimizely/event/event_processor.py [hunk 3]
```diff
@@ -60,7 +61,8 @@ def __init__(self,
                 event_queue=None,
                 batch_size=None,
                 flush_interval=None,
-                timeout_interval=None):
+                timeout_interval=None,
+                notification_center=None):
```
Comment (aliabbasrizvi) [inline optimizely/event/event_processor.py]: Can you add some good explanations for all the fields here. Something like: https://github.com/optimizely/python-sdk/blob/master/optimizely/optimizely.py#L42-L59

File: optimizely/event/event_processor.py [hunk 4]
```diff
@@ -59,7 +61,22 @@ def __init__(self,
...
+    """ EventProcessor init method to configure event batching.
+    Args:
+      event_dispatcher: Provides a dispatch_event method which if given a URL and params sends a request to it.
+      logger: Provides a log method to log messages. By default nothing would be logged.
+      default_start: Optional boolean param which starts the consumer thread if set to True.
```
Comment (aliabbasrizvi) [inline optimizely/event/event_processor.py]: Not entirely sure on this variable name. How about `start_on_init`?

File: optimizely/event/event_processor.py [hunk 5]
```diff
@@ -59,7 +61,22 @@ def __init__(self,
...
+                     By default thread does not start unless 'start' method is called.
+      event_queue: Optional component which accumulates the events until dispacthed.
+      batch_size: Optional param which defines the upper limit of the number of events in event_queue after which
+                  the event_queue will be flushed.
+      flush_interval: Optional param which defines the time in milliseconds after which event_queue will be flushed.
+      timeout_interval: Optional param which defines the time in milliseconds before joining the consumer
```
Comment (aliabbasrizvi) [inline optimizely/event/event_processor.py]: I'd like for us to have any time intervals in seconds like we already have for datafile management.

File: optimizely/event/event_processor.py [hunk 6]
```diff
@@ -195,6 +200,12 @@ def _flush_queue(self):
 
     log_event = EventFactory.create_log_event(to_process_batch, self.logger)
 
+    if self.notification_center is not None:
```
Comment (aliabbasrizvi) [inline optimizely/event/event_processor.py]: Should we be validating that `notification_center` is an instance of `notification_center.NotificationCenter` or does that happen elsewhere?

File: tests/test_event_processor.py [hunk 7]
```diff
@@ -371,3 +375,29 @@ def test_init__NaN_timeout_interval(self):
     # default timeout interval is 5s.
     self.assertEqual(self._event_processor.timeout_interval, timedelta(seconds=5))
     mock_config_logging.info.assert_called_with('Using default value for timeout_interval.')
+
+  def test_notification_center(self):
```
Comment (aliabbasrizvi) [inline tests/test_event_processor.py]: nit. `test_notification_center__on_log_event`

**Pull request merged**